presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilterUtils.java

```diff
@@ -347,7 +347,8 @@ public static TupleDomainFilter toBigintValues(long[] values, boolean nullAllowe
         // slots in a hash table), e.g. up to 192 bits per value.
         // Filter based on a bitmap uses (max - min) / num-values bits per value.
         // Choose the filter that uses less bits per value.
-        if ((max - min + 1) > Integer.MAX_VALUE || ((max - min + 1) / values.length) > 192) {
+        long range = max - min + 1;
+        if (range > Integer.MAX_VALUE || range < Integer.MIN_VALUE || (range / values.length) > 192) {
```

Comment:
Since max >= min, range must be > 0. Hence, `range < Integer.MIN_VALUE` check is not needed. Let's remove.

Reply:
-Long.MIN_VALUE is Long.MIN value, that is why we need the check for < Integer.MIN_VALUE, thats the fix

consider max 10,  min -Long.MIN_VALUE
`10 - Long.MIN_VALUE - 1 = -9223372036854775799`

Reply:
There are other corner cases like Long.MAX - Long.MIN + 1 = 0
For such kind of corner cases using a BigInteger gives us accurate results